### DISCUSSION TITLE
how to use cloudflare context

### DISCUSSION THREAD

**Discussion by andrew-r-thomas** [pbteja1998/remix-auth-email-link#39] [General]
in order to use environment variables in cloudflare, you need access to a context variable. This is a standard part of loaders and actions, and is implemented in the FormStrategy strategy, but it is not for email links, is there a plan to add this in the future/does anyone know how to use cloudflare context with this strategy as is

**Comment by pbteja1998** [comment]
You can use context here too. No need for env variables to be on global process object. 

Maybe I’m missing something here. Can you point me to the specific code where you are facing problems?

**Comment by andrew-r-thomas** [comment]
its possible that I am just new to remix and don't understand some of the subtleties, could you provide an example of how I could say load the magic link secret from context rather than process

**Reply by pbteja1998** [reply]
Ok, I think I now understand what you mean. You are using Cloudflare Pages instead of Cloudflare Workers.

I use Cloudflare Workers, so I have the env variables as global variables.

In your case, you may have to hardcode the secret in the `auth.server.ts` file. 

Since `auth.server.ts` is a server-only file, users won't be able to see the secret, so I think it should be fine even if you hard-code it.

What do you think?

**Reply by andrew-r-thomas** [reply]
I'd rather not mostly because I don't want to check a secret into git, would it be alright if I submit a pull request to add support for cloudflare pages?

is there anything else you can think of that would be necessary to support pages? I'd be happy to work on it

**Reply by pbteja1998** [reply]
If you already have a solution for this, then sure, please go ahead and open a PR. Happy to look into your implementation.

**Comment by E-Kuerschner** [comment]
@andrew-r-thomas did you ever get your PR up? I'm running into the same issue as you. I got around it by creating the instance of my EmailLinkStrategy inside the getLoadContext function

```export const onRequest = createPagesFunctionHandler({
  build,
  mode: build.mode,
  getLoadContext: (context: WorkerContextType) => {
    return {
      env: context.env,
      waitUntil: context.waitUntil,
      log: createLogger(context.env),
      db: getDBClient(context.env),
      auth: createAuthService(context.env),
    };
  },
});```
